src/utils/validation.ts

```diff
@@ -2,9 +2,8 @@ import chains from '@/config/chains'
 import { parsePrefixedAddress, sameAddress, isChecksummedAddress } from './addresses'
 import { safeFormatUnits, safeParseUnits } from './formatters'
 
+export const ADDRESS_RE = /^0x[0-9a-f]{40}$/i
```

Comment:
Then we should probably split `validateAddress` function because it is doing two things at the moment:
- validating the address format
- validating if checksum

Initially I was using that function but it would filter out any address not checksummed and I think we don't want to go that strict in the AB, right?

Reply:
> If we need a helper that returns a boolean

I think we can use the validation as it is -> returning `undefined` if passed all checks

Reply:
We checksum every address which gets added to the address book. And if its incorrectly checksummed it will not let you add that address. So I think we should go quite strict. 
Some APIs only work with checksummed addresses including our tx service for instance.

Reply:
If we validate them when added to the AB, _and_ in address inputs, validating also in the selector seems like an overkill. And it's computationally not the fastest.

Reply:
I've included the checksum validation. Reason being an unchecksummed address might be introduced during a development stage, the same way the corrupted AB with an undefined entry was.

Reply:
> And it's computationally not the fastest.

I didn't experience any degradation of performance but to be fair my AB is not extensively populated.